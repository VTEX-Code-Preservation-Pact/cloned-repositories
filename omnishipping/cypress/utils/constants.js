export const BASE_URL = 'http://vtexgame1.vtexlocal.com.br:3000/'
export const BASE_URL_COMPLETE = 'http://vtexgame1.vtexlocal.com.br/'
export const ADD_SKUS_ENDPOINT = 'checkout/cart/add?'
export const CHECKOUT_ENDPOINT = 'checkout/'
export const PROFILE_ENDPOINT = 'api/checkout/pub/profiles/'
