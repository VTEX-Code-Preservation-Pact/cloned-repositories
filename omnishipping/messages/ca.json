{
  "omnishipping.deliveryTitle": "Entrega",
  "omnishipping.loading": "Carrgant…",
  "omnishipping.awatingData": "Encara falten dades per completar",
  "omnishipping.deliveryAddressTitle": "Completa la teva adreça d'entrega",
  "omnishipping.paymentCTA": "Anar al pagament",
  "omnishipping.locatePickupPoint": "Cercar un punt",
  "omnishipping.deliver": "Rebre",
  "omnishipping.pickup": "Recollir",
  "omnishipping.deliverySummary": "Rebre",
  "omnishipping.package": "Paquet",
  "omnishipping.items": "{itemsAmount, plural, =0 {} one {# producte} other {# productes}}",
  "omnishipping.pickupSummary": "Recollida",
  "omnishipping.cancelBackList": "Consultar altres punts de recollida",
  "omnishipping.cancelPickup": "Cancel·lar la recollida",
  "omnishipping.deliveryOption": "Opcions d'entrega",
  "omnishipping.pickupOption": "Forma de retirada",
  "omnishipping.chooseDelivery": "Rebre",
  "omnishipping.choosePickup": "Recollida a",
  "omnishipping.findPickup": "Trobar un punt de recollida proper.",
  "omnishipping.noPickupSelected": "No hi ha una botiga seleccionada",
  "omnishipping.pickupReceiver": "Responsable de la retirada",
  "omnishipping.slaOption": "{name} - {price} - {shippingEstimate}",
  "omnishipping.slaOptionSimple": "{shippingEstimate}{scheduled}",
  "omnishipping.shippingEstimateScheduled": "Data programada",
  "omnishipping.shippingEstimate-bd": "{timeAmount, plural, =0 {El mateix dia} one {En # dia hábil} other {Fins a # dies hàbils}}",
  "omnishipping.shippingEstimate-db": "Dies laborables",
  "omnishipping.shippingEstimate-d": "{timeAmount, plural, =0 {El mateix dia} one {En # dia} other {Fins a # dies}}",
  "omnishipping.shippingEstimate-h": "{timeAmount, plural, =0 {Amb recollida immediata} one {En # hora} other {Fins a # hores}}",
  "omnishipping.shippingEstimate-m": "{timeAmount, plural, =0 {Amb recollida immediata} one {En # minut} other {Fins a # minuts}}",
  "omnishipping.free": "Gratis",
  "omnishipping.shippingEstimatePickup-bd": "{timeAmount, plural, =0 {A punt en el mateix dia} one {A punt fins # dies hàbils} other {Listo en hasta # días hábiles}}",
  "omnishipping.shippingEstimatePickup-d": "{timeAmount, plural, =0 {A punt en el mateix dia} one {A punt fins # dia} other {A punt a fins # dies}}",
  "omnishipping.shippingEstimatePickup-h": "{timeAmount, plural, =0 {Amb recollida immediata} one {A punt fins # hora} other {A punt a fins # hores}}",
  "omnishipping.shippingEstimatePickup-m": "{timeAmount, plural, =0 {Amb recollida immediata} one {A punt fins # minut} other {A punt a fins # minuts}}",
  "omnishipping.scheduled": "Programada",
  "omnishipping.scheduledDeliveryChoice": "Escull una data d'entrega",
  "omnishipping.scheduledDeliverySplitDisclaimer": "Alguns Itenes no es poden programar. Seleccioneu una forma de lliurament per a rebre'ls.",
  "omnishipping.scheduledDeliveryPartial": "{amount, plural, =0 { } one {Programar # product} other {Programar # productes}}",
  "omnishipping.scheduledDeliveryPrice": "A partir de {price}",
  "omnishipping.scheduledDeliveryAll": "Programar tots",
  "omnishipping.deliveryOptionsLeft": "Forma d'entrega per als productes restants",
  "omnishipping.selectPickupPoint": "Esculli un punt de recollida",
  "omnishipping.selectDeliveryOptions": "Seleccioneu les formes d'entrega",
  "omnishipping.pointDetails": "Detalls del punt de recollida",
  "omnishipping.closeButton": "Tancar",
  "omnishipping.deliveryDate": "Programació",
  "omnishipping.deliveryTime": "Hora d'entrega",
  "omnishipping.datepickerDateFormat": "dddd, LL",
  "omnishipping.deliveryWindows": "Des de {startDate} fins {endDate}",
  "omnishipping.deliveryWindowsSummary": "{date}, des de les {startDate} fins {endDate}",
  "omnishipping.deliveryWindowsSummaryDate": "{date}, Entre {startDate} i {endDate}",
  "omnishipping.deliveryWindowsSummarytime": "Des de {startDate} fins a {endDate}",
  "omnishipping.list": "Llista",
  "omnishipping.map": "Mapa",
  "omnishipping.pickupPoint.modify": "Canviar",
  "omnishipping.modify": "canviar",
  "omnishipping.modifyPostalCode": "canviar",
  "omnishipping.details": "Veure més detalls",
  "omnishipping.change": "Veure tots els punts de recollida",
  "omnishipping.searchLocationMap": "Cercar per Lloc, Codi Postal, Barri o Ciutat",
  "omnishipping.productsInPoint": "Productes disponibles en aquest punt",
  "omnishipping.aditionalInfo": "Altres informacions",
  "omnishipping.unavailable": "No disponible",
  "omnishipping.unavailableItemsAmount": "{itemsAmount, plural, =0 {} one {# No disponible} other {# no estan disponibles}}",
  "omnishipping.confirmPoint": "Recollida en aquest punt",
  "omnishipping.modifyDeliveryOptions": "Canviar opcions d'entrega",
  "omnishipping.shippingAddress": "Adreça d'entrega",
  "omnishipping.editSelectedAddressButton": "Modificar l'adreça seleccionada",
  "omnishipping.addDeliveryPlaceButton": "Entregar en una altra adreça",
  "omnishipping.unavailableDelivery": "{items, plural, =0 {} one {El següent producte està no té entrega disponible per al codi postal escollit.} other {Els següents productes no tenen entrega disponible per al codi postal escollit.}}",
  "omnishipping.unavailableDeliveryAvailablePickup": "{items, plural, =0 {} one {El següent element no està disponible per al lliurament. Podeu treure-ho en un moment.} other {Els següents elements no estan disponibles per al lliurament. Podeu treure-les en un punt.}}",
  "omnishipping.unavailableDeliveryOptionsPostalCode": "Canvieu el codi postal o {removeFromCart}",
  "omnishipping.unavailableDeliveryOptions": "Seleccioneu un altre punt de recollida o {removeFromCart}",
  "omnishipping.changePostalCode": "Canvieu el seu codi postal",
  "omnishipping.optionPackagesDetails": "{items, plural, =0 {...} one {{shippingEstimate}} other {{items} paquets en diferents terminis}}",
  "omnishipping.onePackageIn": "Un paquet a",
  "omnishipping.removeFromCart": "{items, plural, =0 {} one {retira el producte de la cistella.} other {retira els productes de la cistella.}}",
  "omnishipping.pickUpItems": "{items, plural, =0 {} one {Retira l'element en un punt} other {Retira els elements en un punt}}",
  "omnishipping.allItemsPickup": "Només hi ha disponible l'opció Take Out per als vostres articles.",
  "omnishipping.noTimesAvailable": "No hi ha horari disponible per a aquesta data.",
  "omnishipping.productsFromSeller": "Productes de {seller}",
  "omnishipping.fillDeliveryData": "Ompliu el lliurament",
  "omnishipping.deliverySplitDisclaimer": "Alguns productes no poden ser entregats en aquesta direcció. Seleccioneu un punt de recollida.",
  "omnishipping.deliverySplitDisclaimerLogged": "Alguns productes no estan disponibles per a la seva direcció. Recull en un punt.",
  "omnishipping.pickupSplitDisclaimer": "Alguns productes no estan disponibles en el punt seleccionat. Introduïu la vostra adreça per rebre'ls.",
  "omnishipping.pickupSplitDisclaimerLogged": "Alguns productes no estan disponibles en el punt seleccionat. Introduïu la vostra adreça per rebre'ls.",
  "omnishipping.price": "{value, plural, =0 {Gratis} other {{formattedPrice}}}",
  "omnishipping.of": "de",
  "omnishipping.askGeolocation": "Trobi una adreça amb la seva ubicació",
  "omnishipping.cheapestOption": "Més econòmica",
  "omnishipping.fastestOption": "Més ràpida",
  "omnishipping.combinedOption": "Combinada",
  "omnishipping.yourPackages": "Seus paquets",
  "omnishipping.pickupSomeItems": "1. Seleccioneu Opcions de retirada",
  "omnishipping.receiveRemainingPackages": "2. Rebre els productes restants",
  "omnishipping.askGeolocationTitle": "Suggerirem un punt de recollida a prop seu",
  "omnishipping.askGeolocationSubtitle": "Per això, necessitem saber la seva ubicació",
  "omnishipping.askGeolocationAccept": "Utilitza la meva ubicació actual",
  "omnishipping.askGeolocationManual": "Cercar un punt manualment",
  "omnishipping.askGeolocationDenied": "Cercar per una adreça",
  "omnishipping.askGeolocationDeniedTitle": "Trobi punts de recollida propers",
  "omnishipping.askGeolocationDeniedSubtitle": "Busqui les adreces que vostè freqüenta i localitzarem els punts propers",
  "omnishipping.askGeolocationWaiting": "Esperant la seva autorització",
  "omnishipping.askGeolocationInstructions": "Seleccioneu \" Permet \" en l'alerta que apareix al seu navegador",
  "omnishipping.askGeolocationSearching": "Buscant la seva ubicació",
  "omnishipping.backToAddressList": "Tornar a la llista d'adreces",
  "omnishipping.jsError": "Ha succeït un conflicte inesperada",
  "omnishipping.reloadThePage": "Carregar la pàgina",
  "omnishipping.errorId": "ErrorId: {errorId}",
  "omnishipping.noAddressTitle": "Dades de l'adreça incompletes",
  "omnishipping.noAddressCTA": "Per introduir una adreça, {action}.",
  "omnishipping.noAddressAction": "Confirmi la seva identitat",
  "omnishipping.deliverComplement": "a casa",
  "omnishipping.pickupComplement": "a una botiga"
}
