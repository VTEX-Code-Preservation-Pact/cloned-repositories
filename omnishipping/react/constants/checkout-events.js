export const ENABLE = 'enable.vtex'
export const DISABLE = 'disable.vtex'
export const COMPONENT_VALIDATED = 'componentValidated.vtex'
export const COMPONENT_DONE = 'componentDone.vtex'
export const SHIPPING_INFO_UPDATED = 'shippingInformationUpdated.vtex'
export const ORDER_FORM_UPDATED = 'orderFormUpdated.vtex'
export const SHOW_MESSAGE = 'showMessage.vtex'
export const DELIVERY_SELECTED = 'deliverySelected.vtex'
export const AUTHENTICATED_USER = 'authenticatedUser.vtexid'
export const CLOSED_VTEXID = 'closed.vtexid'
