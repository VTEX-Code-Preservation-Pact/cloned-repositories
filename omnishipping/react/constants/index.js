export const PICKUP_IN_STORE = 'pickup-in-point'
export const DELIVERY = 'delivery'
export const PICKUP = 'pickup'
export const SCHEDULED_DELIVERY = 'Entrega Agendada'

export const RESIDENTIAL = 'residential'
export const COMMERCIAL = 'commercial'
export const SEARCH = 'search'
export const GIFT_REGISTRY = 'giftRegistry'

export const GEOCORDS = 'geoCoords'
export const ACCOUNT_INFO_ENDOPOINT =
  '/api/logistics/pub/shipping/configuration'
export const HIDE_MAP = 'HIDE_MAP'
export const SHOW_MAP = 'SHOW_MAP'
export const AVAILABLE = 'available'
export const SHIPPING_HASH = '#/shipping'
export const PAYMENT_HASH = '#/payment'

export const CHEAPEST = 'CHEAPEST'
export const FASTEST = 'FASTEST'
export const COMBINED = 'COMBINED'

export const DENIED = 'denied'

export const VTEXLOCAL = 'vtexlocal'
export const VTEXBETA = 'vtexcommercebeta'
export const VTEXSTABLE = 'vtexcommercestable'

export const DEFAULT_WIDTH = 50
export const DEFAULT_HEIGHT = 50

export const IMPORTANT = 'Important'
export const CRITICAL = 'Critical'
export const ERROR = 'Error'
export const INFO = 'Info'

export const OMNISHIPPING = 'omnishipping'
