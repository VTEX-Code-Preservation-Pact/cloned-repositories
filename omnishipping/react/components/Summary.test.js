import { CHEAPEST, DELIVERY, PICKUP_IN_STORE } from '../constants'
import { setLocale } from 'enzyme-react-intl'

import BRA from '@vtex/address-form/lib/country/BRA'
import ConnecteSummary from './Summary'
import { IntlProvider } from 'react-intl'
import messages from '../../messages/pt.json'
import { Provider } from 'react-redux'
import React from 'react'
import { addValidation } from '@vtex/address-form'
import renderer from 'react-test-renderer'

setLocale('pt')

describe('Summary', () => {
  let state, store, props, onModifyDeliveryOptions
  const address = {
    addressType: 'residential',
    receiverName: null,
    addressId: '10',
    postalCode: '22222222',
    city: 'Rio de janeiro',
    state: 'RJ',
    country: 'BRA',
    street: 'Av das AmÃ©ricas',
    number: '',
    neighborhood: 'Barra da Tijuca',
    complement: 'Loja Barra da Tijuca',
    reference: null,
    geoCoordinates: [],
    addressQuery: 'query',
  }
  const logisticsInfo = {
    itemIndex: 0,
    selectedSla: '3',
    selectedDeliveryChannel: DELIVERY,
    deliveryChannels: [
      {
        id: PICKUP_IN_STORE,
      },
      {
        id: DELIVERY,
      },
    ],
    slas: [
      {
        id: '1',
        deliveryChannel: DELIVERY,
        name: 'test',
        price: 100,
        shippingEstimate: '1bd',
        availableDeliveryWindows: [],
      },
      {
        id: '2',
        deliveryChannel: DELIVERY,
        name: 'test',
        price: 100,
        shippingEstimate: '1bd',
        availableDeliveryWindows: [],
      },
      {
        id: '3',
        deliveryChannel: DELIVERY,
        name: 'test',
        price: 100,
        shippingEstimate: '1bd',
        availableDeliveryWindows: [],
      },
    ],
  }

  beforeEach(() => {
    onModifyDeliveryOptions = jest.fn()
    jest.mock('../utils/Currency', () => 'R$ 1,00')

    state = {
      delivery: {
        activeDeliveryOption: CHEAPEST,
        CHEAPEST: [logisticsInfo, logisticsInfo, logisticsInfo],
      },
      addressForm: {
        residentialId: address.addressId,
        addresses: {
          [address.addressId]: addValidation(address),
        },
      },
      addressRules: {
        rules: {
          BRA,
        },
        selectedRules: BRA,
      },
      orderForm: {
        canEditData: true,
        clientPreferencesData: {
          locale: 'pt-BR',
        },
        items: [
          { seller: '1', imageUrl: '1', name: 'xablau', uniqueId: 'test' },
          { seller: '1', imageUrl: '1', name: 'xablau', uniqueId: 'test' },
          { seller: '1', imageUrl: '1', name: 'xablau', uniqueId: 'test' },
        ],
        sellers: [{ id: '1' }],
        shippingData: {
          logisticsInfo: [logisticsInfo, logisticsInfo, logisticsInfo],
        },
        storePreferencesData: {
          countryCode: 'BRA',
          currencyCode: 'BRL',
          currencySymbol: 'R$',
          currencyFormatInfo: {
            currencyDecimalDigits: 2,
            currencyDecimalSeparator: ',',
            currencyGroupSeparator: '.',
            currencyGroupSize: 3,
            startsWithCurrencySymbol: true,
          },
        },
      },
    }
    store = {
      subscribe: jest.fn(),
      dispatch: jest.fn(),
      getState: jest.fn(() => state),
    }
    props = {
      onModifyDeliveryOptions,
      address,
      sellerId: '1',
      selectedRules: BRA,
      logisticsInfo: {
        selectedSla: '3',
        selectedDeliveryChannel: DELIVERY,
        deliveryChannels: [
          {
            id: PICKUP_IN_STORE,
          },
          {
            id: DELIVERY,
          },
        ],
        slas: [
          {
            name: 'test',
            id: '1',
            deliveryChannel: DELIVERY,
            price: 100,
            shippingEstimate: '1bd',
            availableDeliveryWindows: [],
          },
          {
            name: 'test',
            id: '2',
            deliveryChannel: DELIVERY,
            price: 100,
            shippingEstimate: '1bd',
            availableDeliveryWindows: [],
          },
        ],
      },
      index: 0,
      storePreferencesData: {
        countryCode: 'BRA',
        currencyCode: 'BRL',
        currencySymbol: 'R$',
        currencyFormatInfo: {
          currencyDecimalDigits: 2,
          currencyDecimalSeparator: ',',
          currencyGroupSeparator: '.',
          currencyGroupSize: 3,
          startsWithCurrencySymbol: true,
        },
      },
    }
  })

  it('should render only with DELIVERY and one seller', () => {
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <IntlProvider
            locale="pt"
            messages={{ ...messages, ...{ 'country.BRA': 'Brazil' } }}>
            <ConnecteSummary {...props} rules={BRA} />
          </IntlProvider>
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render only with PICKUP_IN_STORE and one seller', () => {
    const logisticsInfo = {
      itemIndex: 0,
      selectedSla: '3',
      selectedDeliveryChannel: PICKUP_IN_STORE,
      deliveryChannels: [
        {
          id: PICKUP_IN_STORE,
        },
      ],
      slas: [
        {
          id: '1',
          deliveryChannel: PICKUP_IN_STORE,
          name: 'test',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: {
            friendlyName: 'test',
            address,
          },
        },
        {
          id: '2',
          deliveryChannel: PICKUP_IN_STORE,
          name: 'test',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: {
            friendlyName: 'test',
            address,
          },
        },
        {
          id: '3',
          deliveryChannel: PICKUP_IN_STORE,
          name: 'test',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: {
            friendlyName: 'test',
            address,
          },
        },
      ],
    }

    state = {
      delivery: {
        activeDeliveryOption: CHEAPEST,
        CHEAPEST: [logisticsInfo, logisticsInfo, logisticsInfo],
      },
      addressForm: {
        residentialId: address.addressId,
        addresses: {
          [address.addressId]: addValidation(address),
        },
      },
      addressRules: {
        rules: {
          BRA,
        },
        selectedRules: BRA,
      },
      orderForm: {
        canEditData: true,
        clientPreferencesData: {
          locale: 'pt-BR',
        },
        items: [
          { seller: '1', imageUrl: '1', name: 'xablau', uniqueId: 'test' },
          { seller: '1', imageUrl: '1', name: 'xablau', uniqueId: 'test' },
          { seller: '1', imageUrl: '1', name: 'xablau', uniqueId: 'test' },
        ],
        sellers: [{ id: '1' }],
        shippingData: {
          logisticsInfo: [logisticsInfo, logisticsInfo, logisticsInfo],
        },
        storePreferencesData: {
          countryCode: 'BRA',
          currencyCode: 'BRL',
          currencySymbol: 'R$',
          currencyFormatInfo: {
            currencyDecimalDigits: 2,
            currencyDecimalSeparator: ',',
            currencyGroupSeparator: '.',
            currencyGroupSize: 3,
            startsWithCurrencySymbol: true,
          },
        },
      },
    }
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <IntlProvider
            locale="pt"
            messages={{ ...messages, ...{ 'country.BRA': 'Brazil' } }}>
            <ConnecteSummary {...props} rules={BRA} />
          </IntlProvider>
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render only with DELIVERY and PICKUP and two sellers', () => {
    const logisticsInfo = {
      itemIndex: 0,
      selectedSla: '2',
      selectedDeliveryChannel: PICKUP_IN_STORE,
      deliveryChannels: [{ id: PICKUP_IN_STORE }],
      slas: [
        {
          id: '1',
          deliveryChannel: PICKUP_IN_STORE,
          name: 'test',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: { friendlyName: 'test', address },
        },
        {
          id: '2',
          deliveryChannel: PICKUP_IN_STORE,
          name: 'test-1',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: { friendlyName: 'test', address },
        },
        {
          id: '3',
          deliveryChannel: DELIVERY,
          name: 'test-2',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: { friendlyName: null, address: null },
        },
      ],
    }

    const logisticsInfo2 = {
      itemIndex: 1,
      selectedSla: '3',
      selectedDeliveryChannel: DELIVERY,
      deliveryChannels: [{ id: DELIVERY }, { id: PICKUP_IN_STORE }],
      slas: [
        {
          id: '1',
          deliveryChannel: PICKUP_IN_STORE,
          name: 'test',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: { friendlyName: 'test', address },
        },
        {
          id: '2',
          deliveryChannel: PICKUP_IN_STORE,
          name: 'test-1',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: { friendlyName: 'test', address },
        },
        {
          id: '3',
          deliveryChannel: DELIVERY,
          name: 'test-2',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: { friendlyName: null, address: null },
        },
      ],
    }

    const logisticsInfo3 = {
      itemIndex: 2,
      selectedSla: '3',
      selectedDeliveryChannel: DELIVERY,
      deliveryChannels: [{ id: DELIVERY }, { id: PICKUP_IN_STORE }],
      slas: [
        {
          id: '1',
          deliveryChannel: PICKUP_IN_STORE,
          name: 'test',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: { friendlyName: 'test', address },
        },
        {
          id: '2',
          deliveryChannel: PICKUP_IN_STORE,
          name: 'test-1',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: { friendlyName: 'test', address },
        },
        {
          id: '3',
          deliveryChannel: DELIVERY,
          name: 'test-2',
          price: 100,
          shippingEstimate: '1bd',
          availableDeliveryWindows: [],
          pickupStoreInfo: { friendlyName: null, address: null },
        },
      ],
    }

    state = {
      delivery: {
        activeDeliveryOption: CHEAPEST,
        CHEAPEST: [logisticsInfo, logisticsInfo2, logisticsInfo3],
      },
      addressForm: {
        residentialId: address.addressId,
        addresses: {
          [address.addressId]: addValidation(address),
        },
      },
      addressRules: {
        rules: {
          BRA,
        },
        selectedRules: BRA,
      },
      orderForm: {
        canEditData: true,
        clientPreferencesData: {
          locale: 'pt-BR',
        },
        items: [
          { seller: '1', imageUrl: '1', name: 'xablau', uniqueId: 'test' },
          { seller: '1', imageUrl: '1', name: 'xablau', uniqueId: 'test' },
          { seller: '2', imageUrl: '1', name: 'xablau', uniqueId: 'test' },
        ],
        sellers: [{ id: '1' }, { id: '2' }],
        shippingData: {
          logisticsInfo: [logisticsInfo, logisticsInfo2, logisticsInfo3],
        },
        storePreferencesData: {
          countryCode: 'BRA',
          currencyCode: 'BRL',
          currencySymbol: 'R$',
          currencyFormatInfo: {
            currencyDecimalDigits: 2,
            currencyDecimalSeparator: ',',
            currencyGroupSeparator: '.',
            currencyGroupSize: 3,
            startsWithCurrencySymbol: true,
          },
        },
      },
    }

    props = {
      onModifyDeliveryOptions,
      address,
      sellerId: '1',
      deliveryChannel: DELIVERY,
      selectedRules: BRA,
      index: 0,
      storePreferencesData: {
        countryCode: 'BRA',
        currencyCode: 'BRL',
        currencySymbol: 'R$',
        currencyFormatInfo: {
          currencyDecimalDigits: 2,
          currencyDecimalSeparator: ',',
          currencyGroupSeparator: '.',
          currencyGroupSize: 3,
          startsWithCurrencySymbol: true,
        },
      },
    }
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <IntlProvider
            locale="pt"
            messages={{ ...messages, ...{ 'country.BRA': 'Brazil' } }}>
            <ConnecteSummary {...props} rules={BRA} />
          </IntlProvider>
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
