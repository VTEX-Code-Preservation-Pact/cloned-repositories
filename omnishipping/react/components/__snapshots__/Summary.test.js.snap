// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary should render only with DELIVERY and PICKUP and two sellers 1`] = `
Array [
  <div
    className="shp-summary-group group"
>
    <p
        className="shp-summary-group-title title"
    >
        Receber
    </p>
    <div
        className="shp-summary-group-content content"
    >
        <div
            className="shp-summary-group-info info"
        >
            <div
                className="shp-summary-group-address address"
            >
                <div
                    className="address-summary address-summary-BRA"
                >
                    <span>
                        <span
                            className="street"
                        >
                            Av das Américas
                        </span>
                    </span>
                    <span>
                        
                        <span
                            className="complement"
                        >
                            Loja Barra da Tijuca
                        </span>
                    </span>
                    <br
                        className="line1-delimiter"
                    />
                    <span>
                        <span
                            className="neighborhood"
                        >
                            Barra da Tijuca
                        </span>
                        <span
                            className="neighborhood-delimiter-after"
                        >
                             - 
                        </span>
                    </span>
                    <span>
                        <span
                            className="city"
                        >
                            Rio de janeiro
                        </span>
                    </span>
                    <span>
                        <span
                            className="state-delimiter"
                        >
                             - 
                        </span>
                        <span
                            className="state"
                        >
                            RJ
                        </span>
                    </span>
                    <br
                        className="line2-delimiter"
                    />
                    <span>
                        <span
                            className="postalCode"
                        >
                            22222222
                        </span>
                    </span>
                    <br
                        className="line3-delimiter"
                    />
                    <span
                        className="country"
                    >
                        Brazil
                    </span>
                </div>
            </div>
            <div
                className="summaryPackage shp-summary-package"
            >
                <div
                    className="packageTitle shp-summary-package-title"
                >
                    Pacote
                     
                    1
                </div>
                <span
                    className="shp-summary-package-items"
                >
                    1 item
                     
                </span>
                <span
                    className="estimateMultiple shp-summary-package-time"
                >
                    Em até 1 dia útil
                </span>
            </div>
            <div
                className="summaryPackage shp-summary-package"
            >
                <div
                    className="packageTitle shp-summary-package-title"
                >
                    Pacote
                     
                    2
                </div>
                <span
                    className="shp-summary-package-items"
                >
                    1 item
                     
                </span>
                <span
                    className="estimateMultiple shp-summary-package-time"
                >
                    Em até 1 dia útil
                </span>
            </div>
        </div>
        <div
            className="shp-summary-group-price price"
        >
            R$ 2,00
        </div>
    </div>
</div>,
  <div
    className="shp-summary-group group"
>
    <p
        className="shp-summary-group-title title"
    >
        Retirar
    </p>
    <div
        className="shp-summary-group-content content"
    >
        <div
            className="shp-summary-group-info info"
        >
            <div
                className="shp-summary-group-title address"
            >
                test
            </div>
            <div
                className="shp-summary-group-address address pickupAddress"
            >
                <div
                    className="address-summary address-summary-BRA"
                >
                    <span>
                        <span
                            className="street"
                        >
                            Av das Américas
                        </span>
                    </span>
                    <span>
                        
                        <span
                            className="complement"
                        >
                            Loja Barra da Tijuca
                        </span>
                    </span>
                    <br
                        className="line1-delimiter"
                    />
                    <span>
                        <span
                            className="neighborhood"
                        >
                            Barra da Tijuca
                        </span>
                        <span
                            className="neighborhood-delimiter-after"
                        >
                             - 
                        </span>
                    </span>
                    <span>
                        <span
                            className="city"
                        >
                            Rio de janeiro
                        </span>
                    </span>
                    <span>
                        <span
                            className="state-delimiter"
                        >
                             - 
                        </span>
                        <span
                            className="state"
                        >
                            RJ
                        </span>
                    </span>
                    <br
                        className="line2-delimiter"
                    />
                    <span>
                        <span
                            className="postalCode"
                        >
                            22222222
                        </span>
                    </span>
                    <br
                        className="line3-delimiter"
                    />
                    <span
                        className="country"
                    >
                        Brazil
                    </span>
                </div>
            </div>
            <div
                className="summaryPackage shp-summary-package"
            >
                <span
                    className="shp-summary-package-items"
                >
                    1 item
                     
                </span>
                <span
                    className="estimateMultiple shp-summary-package-time"
                >
                    A partir de 1 dia útil
                </span>
            </div>
        </div>
        <div
            className="shp-summary-group-price price"
        >
            R$ 1,00
        </div>
    </div>
</div>,
  <a
    className="link-change-shipping summaryChange"
    href="#/shipping"
    id="open-shipping"
    onClick={[Function]}
>
    Alterar opções de entrega
</a>,
]
`;

exports[`Summary should render only with DELIVERY and one seller 1`] = `
Array [
  <div
    className="shp-summary-group group"
>
    <div
        className="shp-summary-group-content content"
    >
        <div
            className="shp-summary-group-info info"
        >
            <div
                className="shp-summary-group-address address"
            >
                <div
                    className="address-summary address-summary-BRA"
                >
                    <span>
                        <span
                            className="street"
                        >
                            Av das Américas
                        </span>
                    </span>
                    <span>
                        
                        <span
                            className="complement"
                        >
                            Loja Barra da Tijuca
                        </span>
                    </span>
                    <br
                        className="line1-delimiter"
                    />
                    <span>
                        <span
                            className="neighborhood"
                        >
                            Barra da Tijuca
                        </span>
                        <span
                            className="neighborhood-delimiter-after"
                        >
                             - 
                        </span>
                    </span>
                    <span>
                        <span
                            className="city"
                        >
                            Rio de janeiro
                        </span>
                    </span>
                    <span>
                        <span
                            className="state-delimiter"
                        >
                             - 
                        </span>
                        <span
                            className="state"
                        >
                            RJ
                        </span>
                    </span>
                    <br
                        className="line2-delimiter"
                    />
                    <span>
                        <span
                            className="postalCode"
                        >
                            22222222
                        </span>
                    </span>
                    <br
                        className="line3-delimiter"
                    />
                    <span
                        className="country"
                    >
                        Brazil
                    </span>
                </div>
            </div>
            <div
                className="summaryPackage shp-summary-package"
            >
                <span
                    className="false shp-summary-package-time"
                >
                    Em até 1 dia útil
                </span>
            </div>
        </div>
        <div
            className="shp-summary-group-price price"
        >
            R$ 3,00
        </div>
    </div>
</div>,
  <a
    className="link-change-shipping summaryChange"
    href="#/shipping"
    id="open-shipping"
    onClick={[Function]}
>
    Alterar opções de entrega
</a>,
]
`;

exports[`Summary should render only with PICKUP_IN_STORE and one seller 1`] = `
Array [
  <div
    className="shp-summary-group group"
>
    <p
        className="shp-summary-group-title title"
    >
        Retirar
    </p>
    <div
        className="shp-summary-group-content content"
    >
        <div
            className="shp-summary-group-info info"
        >
            <div
                className="shp-summary-group-title address"
            >
                test
            </div>
            <div
                className="shp-summary-group-address address pickupAddress"
            >
                <div
                    className="address-summary address-summary-BRA"
                >
                    <span>
                        <span
                            className="street"
                        >
                            Av das Américas
                        </span>
                    </span>
                    <span>
                        
                        <span
                            className="complement"
                        >
                            Loja Barra da Tijuca
                        </span>
                    </span>
                    <br
                        className="line1-delimiter"
                    />
                    <span>
                        <span
                            className="neighborhood"
                        >
                            Barra da Tijuca
                        </span>
                        <span
                            className="neighborhood-delimiter-after"
                        >
                             - 
                        </span>
                    </span>
                    <span>
                        <span
                            className="city"
                        >
                            Rio de janeiro
                        </span>
                    </span>
                    <span>
                        <span
                            className="state-delimiter"
                        >
                             - 
                        </span>
                        <span
                            className="state"
                        >
                            RJ
                        </span>
                    </span>
                    <br
                        className="line2-delimiter"
                    />
                    <span>
                        <span
                            className="postalCode"
                        >
                            22222222
                        </span>
                    </span>
                    <br
                        className="line3-delimiter"
                    />
                    <span
                        className="country"
                    >
                        Brazil
                    </span>
                </div>
            </div>
            <div
                className="summaryPackage shp-summary-package"
            >
                <span
                    className="false shp-summary-package-time"
                >
                    A partir de 1 dia útil
                </span>
            </div>
        </div>
        <div
            className="shp-summary-group-price price"
        >
            R$ 3,00
        </div>
    </div>
</div>,
  <a
    className="link-change-shipping summaryChange"
    href="#/shipping"
    id="open-shipping"
    onClick={[Function]}
>
    Alterar opções de entrega
</a>,
]
`;
